src/backend/gl/src/command.rs

```diff
@@ -433,37 +433,37 @@ impl RawCommandBuffer {
     }
 
     fn begin_subpass(&mut self) {
-        // Split processing and command recording due to borrowchk.
-        let (draw_buffers, clear_cmds) = {
-            let state = self.pass_cache.as_ref().unwrap();
-            let subpass = &state.render_pass.subpasses[self.cur_subpass];
-
-            // See `begin_renderpass_cache` for clearing strategy
-
-            // Bind draw buffers for mapping color output locations with
-            // framebuffer attachments.
-            let draw_buffers = if state.framebuffer.is_none() {
-                // The default framebuffer is created by the driver
-                // We don't have influence on its layout and we treat it as single image.
-                //
-                // TODO: handle case where we don't do double-buffering?
-                vec![glow::BACK_LEFT]
-            } else {
-                subpass
-                    .color_attachments
-                    .iter()
-                    .map(|id| glow::COLOR_ATTACHMENT0 + *id as u32)
-                    .collect::<Vec<_>>()
-            };
-
-            let clear_cmds = state
-                .render_pass
-                .attachments
+        let state = self.pass_cache.as_ref().unwrap();
+        let subpass = &state.render_pass.subpasses[self.cur_subpass];
+
+        // See `begin_renderpass_cache` for clearing strategy
+
+        // Bind draw buffers for mapping color output locations with
+        // framebuffer attachments.
+        let draw_buffers = if state.framebuffer.fbos[self.cur_subpass].is_none() {
+            // The default framebuffer is created by the driver
+            // We don't have influence on its layout and we treat it as single image.
+            //
+            // TODO: handle case where we don't do double-buffering?
+            vec![glow::BACK_LEFT]
+        } else {
+            subpass
+                .color_attachments
                 .iter()
-                .zip(state.attachment_clears.iter())
-                .filter_map(|(attachment, clear)| {
+                .enumerate()
+                .map(|(index, _)| glow::COLOR_ATTACHMENT0 + index as u32)
+                .collect::<Vec<_>>()
```

Comment:
technically, this doesn't need to collect either

Reply:
How? We need a slice.

Reply:
does `self.add` really need a slice as opposed to an iterator?